tests/Doctrine/Tests/ORM/Query/ParserTest.php

```diff
@@ -134,6 +134,21 @@ public function invalidMatches()
         ];
     }
 
+    /**
+     * PHP 7.4 would fail with Notice: Trying to access array offset on value of type null.
+     * https://github.com/doctrine/orm/pull/7934
+     *
+     * @expectedException \Doctrine\ORM\Query\QueryException
```

Comment:
Can you please use `$this->expectException(QueryException::class);` in the test code instead of the annotation? Put it before that line of code that is throwing the expected exception and import `Doctrine\ORM\Query\QueryException` with `use`.

Reply:
Good point. Done in 000d923363423476bf4a1f0cd940f31548bdeae0.